client/dashboard/app/queries/domain-dns-records.ts

```diff
@@ -14,4 +21,10 @@ export const domainDnsMutation = ( domainName: string ) =>
 				params.recordsToRemove,
 				params.restoreDefaultARecords
 			),
+		onSuccess: () => {
+			// Invalidate and refetch the DNS records after successful mutation
+			queryClient.invalidateQueries( {
+				queryKey: [ 'domains', domainName, 'domain-dns' ],
+			} );
```

Comment:
We can reuse the query:

```suggestion
			queryClient.invalidateQueries( domainDnsQuery( domainName ) );
```